plugins/history_plugin/history_plugin.cpp

```diff
@@ -526,7 +527,8 @@ namespace eosio {
                         result.block_num = *p.block_num_hint;
                         result.block_time = blk->timestamp;
                         fc::mutable_variant_object r("receipt", receipt);
-                        r("trx", chain.to_variant_with_abi(pt.get_signed_transaction(), abi_serializer::create_yield_function( abi_serializer_max_time )));
+                        auto ptv0 = pt.to_packed_transaction_v0();
+                        r("trx", chain.to_variant_with_abi(ptv0->get_signed_transaction(), abi_serializer::create_yield_function( abi_serializer_max_time )));
```

Comment:
Can we manually construct the variant by converting the `transaction` first and then manually appending the `signatures` and `context_free_data` both here and above? That way we avoid the expensive call to `to_packed_transaction_v0`.

Reply:
Added manual construction, but can't really avoid the variant copy with current `mutable_variant_object` capabilities.